conans/test/functional/graph/version_range_override_test.py

```diff
@@ -53,3 +53,31 @@ def test_override_version_range_outside(self):
                       " by your conanfile to libB/3.0@user/channel", self.t.out)
         self.assertIn("ERROR: Version range '<=2.0' required by 'libC/1.0@user/channel' not valid"
                       " for downstream requirement 'libB/3.0@user/channel'", self.t.out)
+
+
+class VersionRangeOverrideFailTestCase(unittest.TestCase):
+
+    def test(self):
+        # https://github.com/conan-io/conan/issues/7864
+        t = TestClient()
+        t.save({"conanfile.py": GenConanfile()})
+        t.run("create . gtest/1.8.0@PORT/stable")
+        t.run("create . gtest/1.8.1@bloomberg/stable")
+
+        t.save({"conanfile.py": GenConanfile().with_require("gtest/1.8.0@PORT/stable")})
+        t.run("create . intermediate/1.0@PORT/stable")
+
+        t.save({"conanfile.py": GenConanfile().with_requires("intermediate/1.0@PORT/stable")
+                                              .with_build_requires("gtest/1.8.0@PORT/stable")})
```

Comment:
```suggestion
                                              .with_build_requirement("gtest/1.8.0@PORT/stable", force_host_context=True)})
```

This limitation/protection should apply when the `build_requires` belongs to the _build_ context, right?

Reply:
The user issue was with a regular declarative ``build_require`` in the build context. I can add another test with a build_require in the host context, it really doesn't matter much, the overriding is the same for both cases.